pkg/apis/v1/nodeclaim_conversion.go

```diff
@@ -139,16 +140,11 @@ func (in *NodeClaimSpec) convertFrom(ctx context.Context, v1beta1nc *v1beta1.Nod
 			MinValues: v1beta1Requirements.MinValues,
 		}
 	})
-
-	defaultNodeClassGVK := injection.GetNodeClasses(ctx)[0]
-	nodeclassGroupVersion, err := schema.ParseGroupVersion(v1beta1nc.NodeClassRef.APIVersion)
-	if err != nil {
-		return "", err
-	}
+	nodeclasses := injection.GetNodeClasses(ctx)
 	in.NodeClassRef = &NodeClassReference{
 		Name:  v1beta1nc.NodeClassRef.Name,
-		Kind:  lo.Ternary(v1beta1nc.NodeClassRef.Kind == "", defaultNodeClassGVK.Kind, v1beta1nc.NodeClassRef.Kind),
-		Group: lo.Ternary(v1beta1nc.NodeClassRef.APIVersion == "", defaultNodeClassGVK.Group, nodeclassGroupVersion.Group),
+		Kind:  lo.Ternary(v1beta1nc.NodeClassRef.Kind == "", nodeclasses[0].Kind, v1beta1nc.NodeClassRef.Kind),
+		Group: lo.Ternary(v1beta1nc.NodeClassRef.APIVersion == "", nodeclasses[0].Group, strings.Split(v1beta1nc.NodeClassRef.APIVersion, "/")[0]),
```

Comment:
can't this just be the following? Looking at ParseGroupVersion, it looks it stores the correct information, but assumes that if there's no /, it treats the string as the version, not the group. 
```suggestion
		Group: lo.Ternary(v1beta1nc.NodeClassRef.APIVersion == "", defaultNodeClassGVK.Group, fmt.Sprintf("%s/%s", defaultNodeClassGVK.Group.Version, nodeclassGroupVersion.Version),
```

Reply:
This way, we keep the upstream package that has more parsing logic for us, rather than implementing it ourselves

Reply:
Yeah, makes sense. I believe this is what you meant - 
```
Group: lo.Ternary(v1beta1np.Spec.NodeClassRef.APIVersion == "", defaultNodeClassGVK.Group, lo.Ternary(nodeclassGroupVersion.Group == "", nodeclassGroupVersion.Version, nodeclassGroupVersion.Group))
```